cpp/src/replace/clamp.cu

```diff
@@ -338,6 +305,85 @@ std::unique_ptr<column> dispatch_clamp::operator()<struct_view>(column_view cons
+  rmm::mr::device_memory_resource* mr,
+  cudaStream_t stream)
+{
+  // add lo_replace and hi_replace to keys
+  auto matched_column = [&] {
+    auto matched_view              = dictionary_column_view(input);
+    std::unique_ptr<column> result = nullptr;
+    if (lo.is_valid()) {
+      result = dictionary::detail::add_keys(
+        matched_view,
+        make_column_from_scalar(lo_replace, 1, rmm::mr::get_current_device_resource(), stream)
+          ->view(),
+        mr,
+        stream);
+      matched_view = dictionary_column_view(result->view());
+    }
+    if (hi.is_valid()) {
+      result = dictionary::detail::add_keys(
+        matched_view,
+        make_column_from_scalar(hi_replace, 1, rmm::mr::get_current_device_resource(), stream)
+          ->view(),
+        mr,
+        stream);
+    }
+    return result;
+  }();
+  auto matched_view = dictionary_column_view(matched_column->view());
+
+  // get the indexes for lo_replace and for hi_replace
+  auto lo_replace_index = dictionary::detail::get_index(
+    matched_view, lo_replace, rmm::mr::get_current_device_resource(), stream);
+  auto hi_replace_index = dictionary::detail::get_index(
+    matched_view, hi_replace, rmm::mr::get_current_device_resource(), stream);
+
+  // get the closests indexes for lo and for hi
+  auto lo_index = dictionary::detail::get_insert_index(
+    matched_view, lo, rmm::mr::get_current_device_resource(), stream);
+  auto hi_index = dictionary::detail::get_insert_index(
+    matched_view, hi, rmm::mr::get_current_device_resource(), stream);
+
+  // call clamp with the scalar indexes and the matched indices
+  auto matched_indices = matched_view.get_indices_annotated();
+  auto new_indices     = cudf::type_dispatcher(matched_indices.type(),
+                                           dispatch_clamp{},
+                                           matched_indices,
+                                           *lo_index,
+                                           *lo_replace_index,
+                                           *hi_index,
+                                           *hi_replace_index,
+                                           mr,
+                                           stream);
+
+  auto const indices_type = new_indices->type();
+  auto const output_size  = new_indices->size();
+  auto const null_count   = new_indices->null_count();
+  auto contents           = new_indices->release();
+  auto indices_column     = std::make_unique<column>(indices_type,
+                                                 static_cast<size_type>(output_size),
+                                                 std::move(*(contents.data.release())),
+                                                 rmm::device_buffer{0, stream, mr},
+                                                 0);
+
+  // take the keys from the matched column allocated using mr
+  std::unique_ptr<column> keys_column(std::move(matched_column->release().children.back()));
```

Comment:
Very similar to code used in your other PRs. Any possible reuse?

Reply:
I may be able to use the new factory function I created in PR #6327 here
https://github.com/rapidsai/cudf/pull/6327/files#diff-484f4708cb779a2d2db763ce7ae6f9adR130-R133
So if that would get merged first, I could probably use it here.
Otherwise I can fix it up in a later PR.

Reply:
You can merge that PR branch into this one and then add "Merge After Dependencies" and note the dependency and we will merge them in order.